tests/test_kgain_cal.py

```diff
@@ -0,0 +1,220 @@
+"""
+Unit test suite for the calibrate_kgain module.
+
+Frames for unit tests are simulated SCI-size frames made with Gaussian and Poisson 
+noises included. The assumed flux map is a realistic pupil image made from 
+TVAC frames.
+"""
+
+import os
+import pytest
+import warnings
+import numpy as np
+import pandas as pd
+from pathlib import Path
+from astropy.io import fits
+
+import test_check
+from corgidrp import check
+from corgidrp.data import Image, Dataset
+from corgidrp.mocks import (create_default_headers, make_fluxmap_image)
+from corgidrp.calibrate_kgain import (calibrate_kgain, CalKgainException, kgain_params)
+
+######################## function definitions ###############################
+
+def nonlin_coefs(filename,EMgain,order):
+    # filename is the name of the csv text file containing the TVAC nonlin table
```

Comment:
I think this function is already defined in mocks.py, right? If its just the same function, then you can just import it here.

Reply:
Good catch! I had added it to mocks.py from calibrate_nonlin.py. Done